src/Collection.php

```diff
@@ -949,15 +1019,23 @@ public function insertOne(array $document)
     }
 
     /**
-     * Returns information for all indexes in the collection.
+     * Returns information for all indexes for the collection.
      *
      * @see http://docs.mongodb.org/manual/reference/command/listIndexes/
      * @see http://docs.mongodb.org/manual/reference/method/db.collection.getIndexes/
-     * @return Cursor
+     * @return IndexInfoIterator
      */
     public function listIndexes()
     {
-        // TODO
+        $readPreference = new ReadPreference(ReadPreference::RP_PRIMARY);
+        $server = $this->manager->selectServer($readPreference);
+
+        $serverInfo = $server->getInfo();
+        $maxWireVersion = isset($serverInfo['maxWireVersion']) ? $serverInfo['maxWireVersion'] : 0;
+
+        return ($maxWireVersion >= 3)
```

Comment:
Feature detection should compare against the min and max wire versions. See `php_mongo_api_connection_supports_feature()` in the 1.x driver.